### PR TITLE
Move SDL2 and Dear ImGui submodules to CMake find_package.

### PR DESCRIPTION
**Pull request by ScottTodd**
Fixes https://github.com/google/iree/issues/3842

We have two optional graphical application binaries using [SDL2](https://www.libsdl.org/) and [Dear ImGui](https://github.com/ocornut/imgui):
* [vulkan_inference_gui](https://github.com/google/iree/blob/main/iree/samples/vulkan/vulkan_inference_gui.cc)
* [iree-run-module-vulkan-gui](https://github.com/google/iree/blob/main/iree/testing/vulkan/iree-run-module-vulkan-gui-main.cc)

As these applications are optional, we don't need to support Bazel.

We don't need to directly include the source for SDL2 and Dear ImGui as git submodules in IREE itself, so this moves those dependencies to CMake-only libraries found via `find_package`. The libraries can be installed in any way developers want, or via [vcpkg](https://github.com/microsoft/vcpkg).

Tested on Windows using:
```
λ vcpkg install sdl2[vulkan]
λ vcpkg install imgui[vulkan-binding,sdl2-binding]
λ vcpkg integrate install
```

The install command instructed me to add `-DCMAKE_TOOLCHAIN_FILE=D:/dev/tools/vcpkg/scripts/buildsystems/vcpkg.cmake` to my CMake configure command, which sets up the includes. We can integrate more directly with vcpkg and document its usage over time.

### CODE REVIEW

File: iree/testing/vulkan/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +22,21 @@ endif()
 # If Vulkan is not found, omit the target.
 include(FindVulkan)
 if(NOT Vulkan_FOUND)
+  message("Could not find Vulkan, disabling Vulkan GUI programs")
```
Comment (benvanik) [inline iree/testing/vulkan/CMakeLists.txt:25]: do these print all the time? like, will this be our equivalent to tf's "NO CUDA AND WHATEVER INTEL MKL MUMBLE FOUND OMG PANIC but actually just a warning" spam?

Comment (ScottTodd) [reply iree/testing/vulkan/CMakeLists.txt:25]: When `IREE_BUILD_SAMPLES` is enabled (it is by default), yes. I'd like to leave _some_ indication of why targets are not defined. Any ideas on how to make this quieter for people that don't care? Maybe an opt-in for messages like this such as `IREE_DIAGNOSE_CONFIGURATION`?

Comment (ScottTodd) [reply iree/testing/vulkan/CMakeLists.txt:25]: Actually, maybe `VERBOSE` is enough? https://cmake.org/cmake/help/latest/command/message.html I'll try it...

> `VERBOSE`
>     Detailed informational messages intended for project users. These messages should provide additional details that won’t be of interest in most cases, but which may be useful to those building the project when they want deeper insight into what’s happening.

Comment (benvanik) [reply iree/testing/vulkan/CMakeLists.txt:25]: verbose would be nice - we can use that in other places too so the default case would be off and then if you hit issues turn it on - like one should be able to find on stack overflow (https://stackoverflow.com/questions/2670121/using-cmake-with-gnu-make-how-can-i-see-the-exact-commands)

Comment (ScottTodd) [reply iree/testing/vulkan/CMakeLists.txt]: `VERBOSE` seems to work, _after a clean reconfigure_

File: iree/testing/vulkan/CMakeLists.txt [hunk 2]
```diff
@@ -35,13 +48,11 @@ iree_cc_library(
   SRCS
     "vulkan_gui_util.cc"
   DEPS
-    dear_imgui::dear_imgui
-    dear_imgui::impl_sdl
-    dear_imgui::impl_vulkan
+    imgui::imgui
```
Comment (benvanik) [inline iree/testing/vulkan/CMakeLists.txt:51]: could these just be `imgui` and `SDL2`?

**Review by benvanik — APPROVED** [review]

File: iree/testing/vulkan/CMakeLists.txt (hunk 2 above)
Comment (ScottTodd) [reply iree/testing/vulkan/CMakeLists.txt:51]: short answer: no

longer answer: maybe, those targets aren't directly exposed by the installed libraries, but we have some special handling in `iree_cc_library` that is similar: https://github.com/google/iree/blob/a49b97b8759436d0a5ece44689a7ca76ddbf3f30/build_tools/cmake/iree_cc_library.cmake#L80-L82

`...\vcpkg\installed\x64-windows\share\imgui\imgui-config.cmake`:
```
# Create imported target imgui::imgui
add_library(imgui::imgui STATIC IMPORTED)
```

`...\vcpkg\installed\x64-windows\share\sdl2\SDL2Targets.cmake`:
```
# Create imported target SDL2::SDL2
add_library(SDL2::SDL2 SHARED IMPORTED)
```

Comment (benvanik) [reply iree/testing/vulkan/CMakeLists.txt:51]: oh cool - I didn't know that they doubled them up like that - that's awesome then and makes us seem less weird :P

**Pull request merged**